autogpt_platform/backend/backend/blocks/smart_decision_maker.py

```diff
@@ -130,6 +139,124 @@ def _get_tool_graph_metadata(self, node_id: str, graph: Any) -> List[Any]:
 
         return graph_meta
 
+    @staticmethod
+    def _create_block_function_signature(
+        sink_node: Any, links: list[Any]
+    ) -> dict[str, Any]:
+        """
+        Creates a function signature for a block node.
+
+        Args:
+            sink_node: The node for which to create a function signature.
+            links: The list of links connected to the sink node.
+
+        Returns:
+            A dictionary representing the function signature in the format expected by LLM tools.
+
+        Raises:
+            ValueError: If the block specified by sink_node.block_id is not found.
+        """
+        block = get_available_blocks()[sink_node.block_id]
+        if not block:
+            raise ValueError(f"Block not found: {sink_node.block_id}")
+
+        tool_function: dict[str, Any] = {
+            "name": re.sub(r"[^a-zA-Z0-9_-]", "_", block().name).lower(),
+            "description": block().description,
+        }
+
+        properties = {}
+        required = []
+
+        for link in links:
+            sink_block_input_schema = block().input_schema
+            description = (
+                sink_block_input_schema.model_fields[link.sink_name].description
+                if link.sink_name in sink_block_input_schema.model_fields
+                and sink_block_input_schema.model_fields[link.sink_name].description
+                else f"The {link.sink_name} of the tool"
```

Comment:
What is "The {link.sink_name} of the tool" supposed to represents ?

Reply:
This is what is added in the case no description is present. Without a description there is very little that can be done. This just uses the name of the variable in a sentence as a description.